pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/metrics/ApexMetricsComputer.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.apex.metrics;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import net.sourceforge.pmd.lang.apex.ast.ASTMethod;
+import net.sourceforge.pmd.lang.apex.ast.ASTUserClassOrInterface;
+import net.sourceforge.pmd.lang.metrics.AbstractMetricsComputer;
+
+/**
+ * Computes metrics for the Apex framework.
+ *
+ * @author Clément Fournier
+ */
+public class ApexMetricsComputer extends AbstractMetricsComputer<ASTUserClassOrInterface<?>, ASTMethod> {
+
+    static final ApexMetricsComputer INSTANCE = new ApexMetricsComputer();
+
+
+    @Override
+    protected List<ASTMethod> findOperations(ASTUserClassOrInterface<?> node) {
+        List<ASTMethod> candidates = node.findChildrenOfType(ASTMethod.class);
+        List<ASTMethod> result = new ArrayList<>(candidates);
+        for (ASTMethod method : candidates) {
+            if (method.getImage().matches("(<clinit>|<init>|clone)")) {
```

Comment:
Yes, that's interesting. We do not only parse the apex code, but we do compile it a little bit - and these are the generated (static) initializers. Looks like, we should really filter them out earlier, since we remove them already in some of the rules (NcssTypeCount, ApexCRUDViolation). Nothing for this PR, but to keep in mind...